source/tests/system/nifgen_driver_api_tests.cpp

```diff
@@ -426,7 +540,56 @@ TEST_F(NiFgenDriverApiTest, ResetInterchangeCheck_ResetsSuccessfully)
   ::grpc::Status status = GetStub()->ResetInterchangeCheck(&context, request, &response);
 
   EXPECT_TRUE(status.ok());
-  EXPECT_EQ(kfgenDriverApiSuccess, response.status());
+  expect_api_success(response.status());
+}
+
+TEST_F(NiFgenDriverApiTest, SendSoftwareEdgeTrigger_TriggersSuccessfully)
+{
+  ::grpc::ClientContext context;
+  fgen::SendSoftwareEdgeTriggerRequest request;
+  request.mutable_vi()->set_id(GetSessionId());
+  request.set_trigger(fgen::Trigger::TRIGGER_NIFGEN_VAL_SCRIPT_TRIGGER);
+  request.set_trigger_id("ScriptTrigger0");
+  fgen::SendSoftwareEdgeTriggerResponse response;
+  ::grpc::Status status = GetStub()->SendSoftwareEdgeTrigger(&context, request, &response);
+
+  EXPECT_TRUE(status.ok());
+  expect_api_success(response.status());
+}
+
+TEST_F(NiFgenDriverApiTest, AllocateNamedWaveform_WriteNamedWaveformF64_WaveformWrittenSuccessfully)
```

Comment:
There is no explicit "validate" part in the body of this test. I think better option would be to return "response" object from `write_named_waveform_f64` and performing `expect_api_success(response.status());` here instead of helper method. Helper method can still assert `status.ok()` to ensure successful grpc call but validation of actual response should happen here in the test method.